src/node_jsvmapi.cc

```diff
@@ -585,96 +623,235 @@ void napi_set_return_value(napi_env e,
   info->SetReturnValue(v);
 }
 
-napi_propertyname napi_property_name(napi_env e, const char* utf8name) {
+napi_errorvalue napi_property_name(napi_env e, 
+                                   const char* utf8name, 
+                                   napi_propertyname* result) {
   NAPI_PREAMBLE(e);
-  v8::Local<v8::String> namestring =
-      v8::String::NewFromUtf8(v8impl::V8IsolateFromJsEnv(e), utf8name,
-          v8::NewStringType::kInternalized).ToLocalChecked();
-  return reinterpret_cast<napi_propertyname>(
-      v8impl::JsValueFromV8LocalValue(namestring));
+
+  if (!result) {
+    return napi_set_last_error_code(napi_invalid_arg);
+  }
+
+  auto maybe = v8::String::NewFromUtf8(v8impl::V8IsolateFromJsEnv(e), utf8name, 
+    v8::NewStringType::kInternalized);
+
+  // Is this an OOM?
```

Comment:
`Local` has an [empty-check](https://v8docs.nodesource.com/node-7.4/de/deb/classv8_1_1_local.html#a821ace5666d33802aff59f69f72901dd) as well, so **maybe** ;) we should use that.